helix-core/src/test/java/org/apache/helix/messaging/handling/TestResourceThreadpoolSize.java

```diff
@@ -113,6 +113,35 @@ public void TestThreadPoolSizeConfig() {
     }
   }
 
+  @Test
+  public void testBatchMessageThreadPoolSize() throws InterruptedException {
```

Comment:
we need a test with number of resources with group message enabled > threapool size. This is the scenario where current code deadlocks. This test case does not cover that scenario